### DISCUSSION TITLE
Change Interval Recursively using SAMDTimer

### DISCUSSION THREAD

**Discussion by WillPowellUk** [khoih-prog/SAMD_TimerInterrupt#11] [Q&A] [answered]
I understand how you can use SAMD_ISR_Timer::changeInterval() to change the interval on the fly. Is there a way to do the same for SAMDTimer?

I have tried calling SAMDTimer::setInterval() recursively inside the interrupt function, but this seems to have no effect.

Any ideas?
Will

**Comment by khoih-prog** [comment, marked as answer]
You can do that, but outside ISR. Have a look at [**Change_Interval** example](https://github.com/khoih-prog/SAMD_TimerInterrupt/blob/main/examples/Change_Interval/Change_Interval.ino), especially [**Change_Interval.ino#L142**](https://github.com/khoih-prog/SAMD_TimerInterrupt/blob/main/examples/Change_Interval/Change_Interval.ino#L142)

```
ITimer.setInterval(TIMER_INTERVAL_MS * 1000 * (multFactor + 1), TimerHandler);
```

**Reply by WillPowellUk** [reply]
Ok, thanks. I have moved the call for setInterval outside of the ISR and that worked well.
